src/twin/factory_reset.rs

```diff
@@ -170,14 +199,28 @@ impl FactoryReset {
     }
 
     fn factory_reset_status() -> Result<Option<&'static str>> {
-        let Ok(factory_reset_status) = std::fs::read_to_string(factory_reset_status_path!()) else {
-            bail!(
-                "factory reset status file missing: {}",
-                factory_reset_status_path!()
+        let omnect_os_initramfs_json: serde_json::Value = serde_json::from_reader(BufReader::new(
+            File::open(factory_reset_status_path!()).context("open omnect-os-initramfs.json")?,
+        ))
+        .context("parsing factory reset status")?;
+
+        let factory_reset = &omnect_os_initramfs_json["factory-reset"];
+        anyhow::ensure!(factory_reset.is_object(), "factory-reset is not an object");
+
+        let mut factory_reset_status = String::from("");
+        let status = &factory_reset["status"];
+        let error = &factory_reset["error"];
+        if !status.is_null() {
+            factory_reset_status = format!(
+                "{}:{}",
+                status.to_string().trim_matches('"'),
+                error.to_string().trim_matches('"')
             );
-        };
+        }
 
-        match factory_reset_status.trim_end() {
+        debug!("factory_reset_status: {factory_reset_status}");
+        // ToDo more stati
```

Comment:
Imho we should do this in a new PBI "omnet-portal: adapt to factory-resetv2 + fsck status"